### PR TITLE
Refactor TestSnapshot

### LINKED ISSUE #72: Ensure test cases are independent
In an effort to get a binary that "works", we wrote some tests where a test case depends on the state left by previous test cases.

This is problematic for debugging problems and it tends to lead to a lot of test changes when there is a behavior change or you want to insert a case in the middle of the existing ones.

Places that I'm aware of:

- [ ] TestCacheCapacityOperations in https://github.com/kubernetes-sigs/kueue/blob/main/pkg/capacity/capacity_test.go
- [ ] TestCacheWorkloadOperations in https://github.com/kubernetes-sigs/kueue/blob/main/pkg/capacity/capacity_test.go
- [ ] Scheduler suite https://github.com/kubernetes-sigs/kueue/blob/main/test/integration/scheduler/scheduler_test.go

And there are similar situations in the following, but it's more like a single complex test case in each:
- [ ] TestSnapshot in https://github.com/kubernetes-sigs/kueue/blob/main/pkg/capacity/snapshot_test.go
- [ ] TestFIFOQueue in https://github.com/kubernetes-sigs/kueue/blob/main/pkg/queue/queue_test.go

/priority backlog

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: Yuki Iwai <yuki.iwai.tz@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
I refactored tests for the snapshot.

#### Which issue(s) this PR fixes:

Fixes #72,#269

### CODE REVIEW

**Comment by tenzen-y** [comment]
I adapt the test to preemption.

**Comment by tenzen-y** [comment]
This PR is ready for review.
PTAL.

/assign @alculquicondor

File: pkg/cache/snapshot_test.go [hunk 1]
```diff
@@ -45,394 +43,450 @@ func TestSnapshot(t *testing.T) {
...
-			"foofoo": {
-				Name:   "foofoo",
-				Cohort: &wantCohort,
-				RequestableResources: map[corev1.ResourceName]*Resource{
-					corev1.ResourceCPU: {
-						Flavors: []FlavorLimits{
-							{
-								Name: "demand",
-								Min:  100_000,
+			wantSnapshot: Snapshot{
```
Comment (alculquicondor) [inline pkg/cache/snapshot_test.go:288]: maybe you can do

```golang
wantSnapshot: func() Snapshot {
  cohort := .....
  return Snapshot{
  }
}
```

Comment (tenzen-y) [reply pkg/cache/snapshot_test.go:288]: Sounds good.

File: pkg/cache/snapshot_test.go [hunk 2]
```diff
@@ -45,394 +43,450 @@ func TestSnapshot(t *testing.T) {
...
+		wls          []*kueue.Workload
+		wantSnapshot Snapshot
+	}{
+		"empty": {
+			wantSnapshot: Snapshot{
+				ClusterQueues:   map[string]*ClusterQueue{},
+				ResourceFlavors: map[string]*kueue.ResourceFlavor{},
+			},
+		},
+		"clusterQueues": {
```
Comment (alculquicondor) [inline pkg/cache/snapshot_test.go:59]: independent clusterQueues

File: pkg/cache/snapshot_test.go [hunk 3]
```diff
@@ -45,394 +43,450 @@ func TestSnapshot(t *testing.T) {
...
+					"spot": utiltesting.MakeResourceFlavor("spot").
+						MultiLabels(map[string]string{"c": "d", "instance": "spot"}).Obj(),
+					"default": utiltesting.MakeResourceFlavor("default").Obj(),
 				},
 			},
 		},
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "with-preemption",
+		"workloads": {
```
Comment (alculquicondor) [inline pkg/cache/snapshot_test.go:123]: I would just merge this one with the test case above titled "clusterQueues"

File: pkg/cache/snapshot_test.go (hunk 1 above)
Comment (tenzen-y) [reply pkg/cache/snapshot_test.go]: Done.

File: pkg/cache/snapshot_test.go (hunk 3 above)
Comment (tenzen-y) [reply pkg/cache/snapshot_test.go:123]: I'm ok with that.

File: pkg/cache/snapshot_test.go (hunk 2 above)
Comment (tenzen-y) [reply pkg/cache/snapshot_test.go:59]: Done.

File: pkg/cache/snapshot_test.go (hunk 3 above)
Comment (tenzen-y) [reply pkg/cache/snapshot_test.go:123]: Done.

**Comment by tenzen-y** [comment]
@alculquicondor I addressed your comments. PTAL.

**Comment by alculquicondor** [comment]
/lgtm
Thanks!

**Pull request merged**